src/api/requests.ts

```diff
@@ -1,67 +1,49 @@
+import { EMPTY, ReplaySubject, catchError, mergeMap, switchMap } from "rxjs";
+import { fromFetch } from "rxjs/fetch";
+import { fromPromise } from "rxjs/internal/observable/innerFrom";
 import { IKartLapsPerRun, ILapDataset } from "../models/go-kart-types";
 
-// Get all sessions
-export default function getAllRuns(
-  callback: (allRunsJSON: string[]) => void,
-  errorCallback: (error: Error) => void,
-  finallyCallback?: () => void
-) {
-  fetch(`https://go-kart-api.onrender.com/runs`)
-    .then((response) => response.json())
-    .then((res: string[]) => {
-      if (!res) {
-        errorCallback(new Error("Cannot find Run"));
-      } else {
-        callback(res);
-      }
-    })
-    .catch((error: Error) => errorCallback(error))
-    .finally(() => {
-      if (finallyCallback) finallyCallback();
-    });
-}
+export const signalNewFilenameRequest$ = new ReplaySubject<null>(1);
+export const signalNewLapsPerRunRequest$ = new ReplaySubject<string>(1);
```

Comment:
The use of ReplaySubject is quite interesting. Is there a specific reason why you want to keep track of the last value emitted in particular?

Reply:
Ya, Tim suggested it. When I subscribed using just Subject the data would not show up unless I used ReplaySubject.